### PR TITLE
Alternative Symfony 3.3+ bootstrap adapter support

### PR DESCRIPTION
**Pull request by dave-redfern**

### CODE REVIEW

File: Bootstraps/AbstractSymfony.php [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+
+            //Doctrine
+            //Doctrine\Bundle\DoctrineBundle\DataCollector\DoctrineDataCollector
+            if ($profiler->has('db')) {
+                Utils::bindAndCall(function () {
+                    //$logger: \Doctrine\DBAL\Logging\DebugStack
+                    foreach ($this->loggers as $logger){
+                        Utils::hijackProperty($logger, 'queries', []);
+                    }
+                }, $profiler->get('db'), null, 'Symfony\Bridge\Doctrine\DataCollector\DoctrineDataCollector');
```
Comment (dzubchik) [inline Bootstraps/AbstractSymfony.php]: Why not to use ::class?

Comment (dave-redfern) [reply Bootstraps/AbstractSymfony.php]: It's from the original Symfony bootstrap adapter. I believe it was written that way to handle cases where Doctrine is not loaded.

**Comment by marcj** [comment]
Can you write some sentences about that PR? Why do we need an alternative?

**Comment by andig** [comment]
@marcj see https://github.com/php-pm/php-pm-httpkernel/pull/69 this should also support symfony4

**Comment by marcj** [comment]
It doesn't as it uses still `session.storage.metadata_bag` service, which will fail in Symfony 4.

**Comment by marcj** [comment]
This PR fails on so many levels. I'll use the locateApplicationKernel part (which doesn't work for my application, as it simply returns `Kernel`) and add the idea behind it on my own. Thanks anyways!

**Comment by marcj** [comment]
Ok, the idea behind `composer.json>autoload>psr-4` does not work out.

**Comment by dave-redfern** [comment]
@marcj I would have been happy to update the PR / rebase with any changes you wanted.

The kernel locator code could have been updated to add the class_exists(\AppKernel) / \App\Kernel as the fallback. Right now, there's no ability to use a custom namespace for the kernel. Maybe it could be injected from the ppm.json file instead?

**Pull request closed without merging**